CONTRIBUTING.md

```diff
@@ -0,0 +1,206 @@
+- **If the problem wasn't triggered by a specific action, describe what you were doing before the problem happened** and share more information using the guidelines below.
+
+### **Suggesting Enhancements ✨**
+
+This section guides you through submitting an enhancement suggestion for OurPromise, including completely new features and minor improvements to existing functionality. Following these guidelines helps maintainers and the community understand your suggestion and find related suggestions 🔎.
+
+Before creating enhancement suggessions, please check if its already existing or not. When you are creating an enhancement suggestion, please include as many details as possible. Just include the steps that you imagine you would take if the feature you're requesting existed.
+
+**Before Submitting An Enhancement Suggestion**
+
+- You might discover that the enhancement is already available.
+- Check if there's a package which provides the same.
+- Check if it was already requested.
+
+**How To Submit A Good Enhancement Suggestion?**
+
+Enhancement suggestions are tracked as [GitHub issues](https://guides.github.com/features/issues/). After deciding your enhancement create an issue in the issues section.
+
+- **Use a clear and descriptive title** for the issue to identify the suggestion.
+- **Provide a step-by-step description of the suggested enhancement** in as many details as possible.
+- **Provide specific examples to demonstrate the steps**. Include copy/pasteable snippets which you use in those examples, as [Markdown code blocks](https://help.github.com/articles/markdown-basics/#multiple-lines).
+- **Describe the current behavior and explain which behavior you expected to see instead** and why.
+- **Include screenshots and animated GIFs** which help you demonstrate the steps or point out the part of the project which the suggestion is related to.
+- **Explain why this enhancement would be useful** to our users.
+
+### **First Code Contribution 🥇**
+
+Unsure on where to begin your contribution? You can start by looking at issues having `Good First Issues`, `Documentation` or `Beginner` tags. You can fine theme [here](https://github.com/kachun333/OurPromise/issues)
+
+### **Pull Request 🥳**
+
+Once you have [forked](https://docs.github.com/en/get-started/quickstart/fork-a-repo) the repo, you need to create your code contributions within a new branch of your forked repo.  
+For general background on creating and managing branches within GitHub, see: [Git Branching and Merging](https://git-scm.com/book/en/v2/Git-Branching-Basic-Branching-and-Merging).
+
+Following are the steps to guide you:
+
+**Step 0:** Clone the forked repo and cd into it
+
+```
+git clone https://github.com/<Your username>/OurPromise.git
+cd OurPromise/
+```
+
+**Step 1:** To begin, create a new branch, give a good [branch name](https://codingsight.com/git-branching-naming-convention-best-practices/). You usually create a branch like so:
```

Comment:
[Thankful 👍]
thanks for sharing the git branch naming convention best practices.
we should follow this